common/bytes.go

```diff
@@ -58,12 +56,38 @@ func CopyBytes(b []byte) (copiedBytes []byte) {
 
 func HasHexPrefix(str string) bool {
 	l := len(str)
-	return l >= 2 && str[0:2] == "0x"
+	return l >= 2 && (str[0:2] == "0x" || str[0:2] == "0X")
+}
+
+func isHexCharacter(c byte) bool {
+	switch {
+	case '0' <= c && c <= '9':
+		return true
+	case 'a' <= c && c <= 'f':
+		return true
+	case 'A' <= c && c <= 'F':
+		return true
+	}
+	return false
 }
 
 func IsHex(str string) bool {
-	l := len(str)
-	return l >= 4 && l%2 == 0 && str[0:2] == "0x"
+	if HasHexPrefix(str) {
+		str = str[2:]
+	}
+
+	if len(str)%2 != 0 {
+		return false
+	}
+
+	src := []byte(str)
+	for _, c := range src {
+		if !isHexCharacter(c) {
+			return false
+		}
+	}
+
+	return true
 }
```

Comment:
Please remove blank lines in this function.

Reply:
done